File: Sesion 7/Marco teorico/4constructores_destructores_clases_derivadas/4.java
import java.util.*;

class Base {
    public Base() {
        System.out.println("Constructor de la clase base");
    }

    public void finalize() {
        System.out.println("Destructor de la clase base");
    }
}

class Derivada extends Base {
    public Derivada() {
        System.out.println("Constructor de la clase derivada");
    }

    public void finalize() {
        System.out.println("Destructor de la clase derivada");
    }
}

public class Main {
    public static void main(String[] args) {
        Derivada obj = new Derivada();
    }
}

File: Sesion 7/Marco teorico/4constructores_destructores_clases_derivadas/4.py
class Base:
    def __init__(self):
        print("Constructor de la clase base")

    def __del__(self):
        print("Destructor de la clase base")

class Derivada(Base):
    def __init__(self):
        super().__init__()
        print("Constructor de la clase derivada")

    def __del__(self):
        super().__del__()
        print("Destructor de la clase derivada")


obj = Derivada()
